spring-cloud-stream-test-support-internal/src/main/java/org/springframework/cloud/stream/test/junit/kafka/KafkaTestSupport.java

```diff
@@ -70,8 +68,14 @@
 	private static boolean hasFailedAlready = false;
 
 	static {
-		// check if either the environment or Java property is set to use embedded tests
-		defaultEmbedded = "true".equals(System.getenv(SCS_KAFKA_TEST_EMBEDDED)) || "true".equals(System.getProperty(SCS_KAFKA_TEST_EMBEDDED));
+		if (System.getProperty(SCS_EXTERNAL_SERVERS_REQUIRED) != null &&
```

Comment:
This feels like a slight abuse of `SCS_EXTERNAL_SERVERS_REQUIRED`, whose role is to forcibly fail tests if the external servers are not present and the rules are not satisfied (thus preventing certain potential failures to get unnoticed). I believe we should reinstate the separate setting for an embedded Kafka server. Thougts?